code/game/objects/structures/morgue.dm

```diff
@@ -29,10 +31,24 @@
 	if(connected)
 		icon_state = "morgue0"
 	else
-		if(contents.len)
-			icon_state = "morgue2"
-		else
-			icon_state = "morgue1"
+		icon_state = "morgue1"
+		var/list/searching = GetAllContents(searchDepth = 3, checkClient = 0) // Search inside bodybags as well.
+		for(var/mob/living/M in searching)
+			if(iscarbon(M))
+				var/mob/living/carbon/C = M
+				if(C.status_flags & FAKEDEATH)
+					icon_state = "morgue2"
+					break
+				switch(C.stat)
+					if(DEAD)
+						icon_state = "morgue2"
+						break
+					if(UNCONSCIOUS)
+						icon_state = "morgue3"
+						break
+					if(CONSCIOUS)
+						icon_state = "morgue4"
+						break
```

Comment:
if you're doing it this way, just break after you find a carbon and do the stuff, you don't need to add it to each line